homeassistant/components/logbook/__init__.py

```diff
@@ -733,14 +746,12 @@ def data_domain(self):
     @property
     def attributes(self):
         """State attributes."""
-        if not self._attributes:
-            if (
-                self._row.attributes is None
-                or self._row.attributes == EMPTY_JSON_OBJECT
-            ):
+        if self._attributes is None:
+            source = self._row.shared_attrs or self._row.attributes
+            if source == EMPTY_JSON_OBJECT or source is None:
                 self._attributes = {}
             else:
-                self._attributes = json.loads(self._row.attributes)
+                self._attributes = json.loads(source)
```

Comment:
Can we have this covered by a test?

Reply:
c4f31a85b8fd0e740807262c2b7e1a494c0f10a4

Reply:
Verified cover